roles/repositories/tasks/main.yml

```diff
@@ -0,0 +1,28 @@
+---
+- name: 'Enable Repositories'
+  theforeman.foreman.repository_set:
+    username: "{{ username }}"
+    password: "{{ password }}"
+    server_url: "{{ server_url }}"
+    validate_certs: "{{ validate_certs | default(omit) }}"
+    organization: "{{ organization }}"
+    product: "{{ item.0.name }}"
+    name: "{{ item.1.name }}"
+    state: enabled
+    repositories:
+      - releasever: "{{ item.1.releasever | default(omit) }}"
+        basearch: "{{ item.1.basearch | default(omit) }}"
+  with_subelements:
+    - "{{ products }}"
+    - repository_sets
+
+- name: 'Synchronize Repositories'
```

Comment:
I'm still not sold that enabling repositories should imply a sync of the repositories.

Reply:
I think, at some point, we mentioned a flag that will result in a sync "always", "never", "if changed".

Honestly, this could also be done with an own role (and give the user better flexibility what they sync). The only caveat is that `import_role` doesn't register a result, so you can't automatically see if something changed by running the role. The role would need to register that itself and we'd have to document that for the user.